update configuration for sass 3.4.14

akoskm opened the pull request:
After updating sass to 3.4.14 grunt task failed with the following message:

```
Running "sass:dev" (sass) task
Error: File to import not found or unreadable: client/bower_components/bootstrap-sass/assets/stylesheets/bootstrap.
        on line 2 of client/src/assets/sass/style.scss
  Use --trace for backtrace.
Warning: Exited with error code 65 Use --force to continue.

Aborted due to warnings.
```

This change supposed to fix the build process.

I added grunt-forever and now `grunt production` should start the application as a daemon.
Stop is possible with SIGKILL or with `grunt forever:drywall:stop`

arthurkao commented:
Hi akoskm,
Thank you for contributing to Angular-Drywall, hope you enjoyed it. 

Regarding to 
- sass 3.4.14 import path issue, I think adding a flag (loadPath) in grunt sass task is a clean(er) alternative to changing the import path to relative path and it's future-proof. I've pushed a fix, please verify.
- grunt-forever in production task is intentionally left out because production task was designed to make all the necessary assets so they are production-ready, instead of running it on production environment. Maybe I should rename it to "buildProduction". Depending on the production environment, there may be built-in forever support already provided at the system level.

Best,
Arthur

akoskm commented:
Hello Arthur,

On Sun, May 31, 2015 at 10:01 AM, Arthur C. Kao notifications@github.com
wrote:

> Hi akoskm,
> Thank you for contributing to Angular-Drywall, hope you enjoyed it.
> 
> Regarding to
> - sass 3.4.14 import path issue, I think adding a flag (loadPath) in
>   grunt sass task is a clean(er) alternative to changing the import path to
>   relative path and it's future-proof. I've pushed a fix, please verify.
> 
> didn't know about the loadPath option, it would be definitely a cleaner
> alternative.
> - grunt-forever in production task is intentionally left out because
>   production task was designed to make all the necessary assets so they are
>   production-ready, instead of running it on production environment. Maybe I
>   should rename it to "buildProduction". Depending on the production
>   environment, there may be built-in forever support already provided at the
>   system level.
> 
> Let me create a runProduction task then which builds the app and starts it
> with grunt-forever in case if there is no built-in forever support (like in
> our case).
> 
> Best,
> Arthur
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/arthurkao/angular-drywall/pull/1#issuecomment-107141894
> .

Thanks,
Akos

## 

Akos Kemives
http://akoskm.github.io/

akoskm commented:
I renamed the production goal to `buildProduction` and added `runProduction` which starts grunt-forever.

arthurkao commented:
Hi akoskm,
Currently I intend not to include grunt forever for production usage. In fact, I argue that the entire grunt business should stay out of the production environment. 

To name a few reasons behind:
- grunt itself and its associating tasks are intended to serve as dev time tool and they could not, and usually would not, be included in your production environment. Refer to package.json devDependencies to see packages used in dev time. These tools are to make our dev time easier, but are not needed to run the project.
- it is best practice NOT to, even though one certainly could, deploy the entire repo to production machine and then build/run on the same directory for security reason(s). After using grunt production task to build production assets, one should only deploy what is really needed (for example, the minified js/css files) to production environment.
- for dev time "forever" task, grunt-nodemon is covering the ground.

Best,
Arthur

akoskm commented:
Arthur,

I really appreciate the detailed reasoning, I agree with you in every point. So far I only used grunt for development purposes but after reading your comment, I'm sure I'll keep this tool out of production environment.

Regarding your second point, could you give me any security reasons? In my opinion what could make the deployment more secure is to do it with a separate user. This user would be designated only to deploy the application and to access the directory which contains the production assets. Is this something you would suggest?

After all this, I see no point in using grunt-forever.

Thanks,
Akos

**EDIT**
I think you can close this merge request without merging anything although I changed the `Gruntfile.js` and added the loadPath modification you suggested.

arthurkao commented:
akoskm,

My two cents:

Only deploy minified (and uglified) assets to the production server. i.e. application.min.js/style.min.css ...etc. Files in client/src/ and non-minified files in client/dist/ are not deployed.

passwords and account information (that are grouped in config.js) are best injected as env variables in production environment, too.

Best,
Arthur
